fivetran_provider_async/operators/fivetran.py

```diff
@@ -0,0 +1,64 @@
+from typing import Any, Dict, Optional
+
+from airflow.exceptions import AirflowException
+from airflow.utils.context import Context
+from fivetran_provider.operators.fivetran import FivetranOperator
+
+from fivetran_provider_async.triggers.fivetran import FivetranTrigger
+
+
+class FivetranOperatorAsync(FivetranOperator):
+    """
+    `FivetranOperatorAsync` submits a Fivetran sync job , and polls for its status on the
+    airflow trigger.`FivetranOperatorAsync` requires that you specify the `connector_id` of
+    the sync job to start. You can find `connector_id` in the Settings page of the connector
+    you configured in the `Fivetran dashboard <https://fivetran.com/dashboard/connectors>`_.
+
+    :param fivetran_conn_id: `Conn ID` of the Connection to be used to configure
+        the hook.
+    :type fivetran_conn_id: str
+    :param connector_id: ID of the Fivetran connector to sync, found on the
+        Connector settings page in the Fivetran Dashboard.
+    :type connector_id: str
+    :param poke_interval: Time in seconds that the job should wait in
+        between each tries
+    :type poke_interval: int
+    :param fivetran_retry_limit: # of retries when encountering API errors
+    :type fivetran_retry_limit: Optional[int]
+    :param fivetran_retry_delay: Time to wait before retrying API request
+    :type fivetran_retry_delay: int
+    """
+
+    def execute(self, context: Dict[str, Any]) -> None:
+        """Start the sync using synchronous hook"""
+        hook = self._get_hook()
+        hook.prep_connector(self.connector_id, self.schedule_type)
+        hook.start_fivetran_sync(self.connector_id)
+
+        """Defer and poll the sync status on the Trigger"""
+        self.defer(
+            timeout=self.execution_timeout,
+            trigger=FivetranTrigger(
+                task_id=self.task_id,
+                fivetran_conn_id=self.fivetran_conn_id,
+                connector_id=self.connector_id,
+                polling_period_seconds=self.poll_frequency,
```

Comment:
Can we keep the `polling_period_seconds` attribute as `poke_interval`? Does sync class both 
`poke_interval` and `poll_frequency ?

Reply:
Changed to use `poke_interval` everywhere. In sync class we have `poll_frequency`, so using that as `poke_interval` now